### PR TITLE
Mongo DB implementation with path filtering

### PR DESCRIPTION
**Pull request by Valecant**
## Proposed changes

The user can now decide to store data inside MongoDB, also the path and the tenant filters are now working

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation (improve documentation)
- [ ] Continuous Integration (functionality that improve automation or testing)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING][contrib] doc
- [x] I have signed the [Contributor License Agreement][cla]
- [x] I have updated the [RELEASE NOTES][release]
- [ ] I have added tests that prove my fix is effective or that my
      feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in
      downstream modules

[cla]: https://raw.githubusercontent.com/orchestracities/auth-management-ui/master/individual_cla.pdf
    "Martel Open Source Software Individual Contributor License Agreement"
[contrib]: https://github.com/orchestracities/auth-management-ui/blob/master/CONTRIBUTING.md
    "Contributing to auth-management-ui"
[release]: https://github.com/orchestracities/auth-management-ui/blob/master/RELEASE_NOTES.md
    "auth-management-ui Release Notes"

### CODE REVIEW

File: configuration-api/main/mongo/alarms.js [hunk 1]
```diff
@@ -1,12 +1,102 @@
 const fs = require('fs');
 const path = require('path');
+const mongoose = require('mongoose');
+const config = require('../config');
+const connection = mongoose.createConnection(config.getConfig().mongo_db);
```
Comment (c0c0n3) [inline configuration-api/main/mongo/alarms.js:5]: what happens if the connection goes down? will we only get a fresh working connection next time we bounce Node? If not, what's going to repair a broken connection?

File: configuration-api/main/mongo/alarms.js [hunk 2]
```diff
@@ -39,6 +129,48 @@ async function updateThisAlarm(data) {
   });
   return [data];
 }
+
+async function getTheAlarms(data) {
```
Comment (c0c0n3) [inline configuration-api/main/mongo/alarms.js:133]: Should we use the factory pattern?

- https://en.wikipedia.org/wiki/Factory_method_pattern

that way we won't need to switch on `operation` in every function...

File: configuration-api/main/advancedAuth.js [hunk 3]
```diff
@@ -77,7 +77,7 @@ const typeDefs = gql`
   type Query {
     listTenants(tenantNames: [String]!): [TenantConfiguration]
     getUserPreferences(userName: String!): [UserPreferencies]
-    getAlarms(tenantName: String!, servicePath: String!): [Alarm]
+    getAlarms(tenantName: String!, servicePath: String!, operation: String!): [Alarm]
```
[the remark this answers is not in the document]
Comment (Valecant) [reply configuration-api/main/advancedAuth.js:80]: The operation is now defined inside the env file

File: configuration-api/main/mongo/alarms.js (hunk 1 above)
Comment (Valecant) [reply configuration-api/main/mongo/alarms.js:5]: GraphQL is managing that, an error on the ui would be displayed and when mongo is back online the app is going to be reloaded automatically

File: configuration-api/main/mongo/alarms.js (hunk 2 above)
Comment (Valecant) [reply configuration-api/main/mongo/alarms.js:133]: the functions are now called based on the name if json or mongo

**Review by c0c0n3 — APPROVED** [review]
great job!

**Pull request merged**